centreon/www/class/centreonGraph.class.php

```diff
@@ -320,13 +320,20 @@ public function __construct($user_id, $index = null, $debug = 0, $compress = nul
             }
             $DBRESULT->closeCursor();
 
-            foreach ($odsm as $mid => $val) {
-                if (!isset($metrics_cache[$mid])) {
-                    $DBRESULT = $this->DB->query(
-                        "INSERT INTO `ods_view_details`
-                            (`metric_id`, `contact_id`, `all_user`, `index_id`)
-                            VALUES ('" . $mid . "', '" . $this->user_id . "', '0', '" . $this->index . "');"
-                    );
+            $insertViewDetailsStatement = $this->DB->prepareQuery(
+                <<<SQL
+                    INSERT INTO `ods_view_details`
+                    (`metric_id`, `contact_id`, `all_user`, `index_id`)
+                    VALUES (:metric_id, :user_id, '0', :index_id)
+                    SQL
+            );
+            foreach (array_keys($odsm) as $mid) {
+                if (! isset($metrics_cache[$mid])) {
+                    $this->DB->executePreparedQuery($insertViewDetailsStatement, [
+                        ':metric_id' => $mid,
+                        ':user_id' => $this->user_id,
+                        ':index_id' => $this->index
+                    ]);
```

Comment:
Catch CentreonDbException and log the error with CentreonLog with business informations and a context please

Reply:
Seen together, you don't want to change your code this time to merge it quickly. 
But the next time, can you catch the exception and use the logger please to improve our error handling ? Thanks.